src/IO/H5/ReadTabulated.hpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class h5::Tabulated
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <hdf5.h>
+#include <string>
+#include <vector>
+
+#include "IO/H5/Object.hpp"
+#include "IO/H5/OpenGroup.hpp"
+
+/// \cond
+class Matrix;
+/// \endcond
+
+namespace h5 {
+/*!
+ * \ingroup HDF5Group
+ *
+ * Contains functions to read in rank-1 and rank-3 datasets from an h5 file.
+ */
+class Tabulated : public h5::Object {
+ public:
+  /// \cond HIDDEN_SYMBOLS
+  static std::string extension() { return ""; }
+
+  Tabulated(bool exists, detail::OpenGroup&& group, hid_t location,
+            const std::string& name);
+
+  Tabulated(const Tabulated& /*rhs*/) = delete;
+  Tabulated& operator=(const Tabulated& /*rhs*/) = delete;
+  Tabulated(Tabulated&& /*rhs*/) noexcept = delete;             // NOLINT
+  Tabulated& operator=(Tabulated&& /*rhs*/) noexcept = delete;  // NOLINT
+
+  ~Tabulated() override;
+  /// \endcond HIDDEN_SYMBOLS
+
+  std::vector<double> get_rank1_dataset() const;
+  std::vector<Matrix> get_rank3_dataset() const;
+
+ private:
+  /// \cond HIDDEN_SYMBOLS
+  std::string name_;
+  std::array<hsize_t, 1> size_1d_;
```

Comment:
This variable is probably unnecessary, but that depends on whether we support writing.  It should be checked when that is settled.